packages/@lwc/engine/src/framework/def.ts

```diff
@@ -162,6 +162,7 @@ function createComponentDef(
         name,
         wire,
         track,
+        fields: undefined,
```

Comment:
this is not needed right? we can remove this from Def completely.

Reply:
@caridy this is needed with your suggested change: `fields: string[] | undefined` `ComponentDef` inherit from `DecoratorsMeta`, and now fields can be string[] or undefined, but should be present.